include/NovelRT/Ecs/Ecs.h

```diff
@@ -10,18 +10,58 @@
  */
 namespace NovelRT::Ecs
 {
-}
+    class Catalogue;
+    class ComponentBufferMemoryContainer;
+    class ComponentCache;
+    class EntityCache;
+    class SparseSetMemoryContainer;
+    class SystemScheduler;
+    class UnsafeComponentView;
+} // namespace NovelRT::Ecs
 
-#include "Catalogue.h"
-#include "ComponentBuffer.h"
+// dependencies for ECS
+// clang-format off
+#include "../Atom.h"
+#include "../Exceptions/Exceptions.h"
+#include "../Timing/Timestamp.h"
+#include "../Utilities/Event.h"
+#include "../Utilities/KeyValuePair.h"
+#include <algorithm>
+#include <atomic>
+#include <cstddef>
+#include <cstdint>
+#include <cstdlib>
+#include <cstring>
+#include <functional>
+#include <iterator>
+#include <memory>
+#include <mutex>
+#include <thread>
+#include <tuple>
+#include <typeindex>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+// clang-format on
+
+// ECS types
+
+// clang-format off
```

Comment:
Nit: why turn on and back off again if nothing is to be parsed here besides 1 comment?

Reply:
Because I wanted that area to be formatted. I could've left it off but I didn't see a need. Not sure I really want to wait for n amount of minutes from CI for this lol